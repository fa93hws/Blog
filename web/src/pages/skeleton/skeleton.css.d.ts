// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'gridBaseline': string;
  'mainContent': string;
  'nav': string;
  'navWidth': string;
}
export const cssExports: CssExports;
export default cssExports;
