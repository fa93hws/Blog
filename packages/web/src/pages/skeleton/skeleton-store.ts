import { observable, action } from 'mobx';

export class SkeletonStore {
  @observable.ref
  isMenuHidden = false;

  @action
  hideMenu() {
    this.isMenuHidden = true;
  }

  @action
  showMenu() {
    this.isMenuHidden = false;
  }
}
