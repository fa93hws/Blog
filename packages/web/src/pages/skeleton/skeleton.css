@value gridBaseline from '~components/metrics.css';

@value navWidth: calc(gridBaseline * 30);

@value animationTransition: 500ms cubic-bezier(.18, .81, .3, .89);

.nav {
  position: fixed;
  width: navWidth;
  display: flex;
  height: 100vh;
  transition: transform animationTransition;
}

.nav.hideMenu {
  transform: translateX(-100%);
}

.mainContent {
  flex-grow: 1;
  padding-left: navWidth;
  transition: padding animationTransition;
}

.mainContent.hideMenu {
  padding-left: 0;
}
