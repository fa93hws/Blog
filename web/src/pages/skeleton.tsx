import * as React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { createHomePage } from './home/home';
import { NavItem } from '../components/nav/nav-links/nav-links';
import { Nav } from '../components/nav/nav';
import { createGlobalHeader } from './global-header/create-golbal-hreader';
import styles from './page.css';

export const Skeleton = ({
  LeftNav,
  MainContent,
  GlobalHeader,
}: {
  LeftNav: React.ComponentType;
  MainContent: React.ComponentType;
  GlobalHeader: React.ComponentType;
}) => (
  <BrowserRouter>
    <div className={styles.container}>
      <aside className={styles.nav}>
        <LeftNav />
      </aside>
      <main className={styles.mainContent}>
        <GlobalHeader />
        <MainContent />
      </main>
    </div>
  </BrowserRouter>
);

export function createSkeleton() {
  const MainContent = createHomePage();
  const author = {
    name: '夏目天子',
    email: 'wjun0912@gmail.com',
    avatar: {
      img: 'https://avatars0.githubusercontent.com/u/10626756',
      alt: "author's github page",
      linkTo: 'https://github.com/fa93hws',
    },
  };
  const navLinkItems: readonly NavItem[] = [
    {
      text: 'Home',
      link: '/home',
    },
    {
      text: 'Archives',
      link: '/archives',
    },
    {
      text: 'Tags',
      link: '/tags',
    },
    {
      text: 'Github',
      link: '/github',
    },
    {
      text: 'Weibo',
      link: '/weibo',
    },
  ];

  const LeftNav = () => (
    <Nav
      author={author}
      navItems={navLinkItems}
      sourceCodeUrl="https://github.com/fa93hws/blog"
    />
  );
  const GlobalHeader = createGlobalHeader();
  const SkeletonImpl = () => (
    <Skeleton
      LeftNav={LeftNav}
      MainContent={MainContent}
      GlobalHeader={GlobalHeader}
    />
  );
  return SkeletonImpl;
}
