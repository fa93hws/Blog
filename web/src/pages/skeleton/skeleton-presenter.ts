import { observable, action } from 'mobx';

export class SkeletonPresenter {
  @observable.ref
  isMenuHidden = false;

  @action
  hideMenu() {
    this.isMenuHidden = true;
  }

  @action
  showMenu() {
    this.isMenuHidden = false;
  }
}
