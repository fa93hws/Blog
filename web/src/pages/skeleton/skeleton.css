@value gridBaseline from '~components/metrics.css';

@value navWidth: calc(gridBaseline * 30);

.nav {
  position: fixed;
  width: navWidth;
  display: flex;
  height: 100vh;
}

.mainContent {
  flex-grow: 1;
  padding-left: navWidth;
}
