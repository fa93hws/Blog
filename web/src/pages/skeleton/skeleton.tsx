import * as React from 'react';
import { observer } from 'mobx-react';
import * as classnames from 'classnames';
import { BrowserRouter } from 'react-router-dom';
import { createHomePage } from 'pages/home/home';
import { NavItem } from './nav/nav-links/nav-links';
import { Nav } from './nav/nav';
import { createHeader } from './header/create-header';
import { SkeletonStore } from './skeleton-store';
import styles from './skeleton.css';

export const Skeleton = ({
  LeftNav,
  MainContent,
  Header,
  isMenuHidden,
}: {
  LeftNav: React.ComponentType;
  MainContent: React.ComponentType;
  Header: React.ComponentType<{ icon: 'Cross' | 'Menu' }>;
  isMenuHidden: boolean;
}) => (
  <BrowserRouter>
    <>
      <aside
        className={classnames(styles.nav, {
          [styles.hideMenu]: isMenuHidden,
        })}
      >
        <LeftNav />
      </aside>
      <main
        className={classnames(styles.mainContent, {
          [styles.hideMenu]: isMenuHidden,
        })}
      >
        <div>
          <Header icon={isMenuHidden ? 'Menu' : 'Cross'} />
          <MainContent />
        </div>
      </main>
    </>
  </BrowserRouter>
);

export function createSkeleton() {
  const MainContent = createHomePage();
  const navLinkItems: readonly NavItem[] = [
    {
      text: 'Home',
      link: '/home',
    },
    {
      text: 'Archives',
      link: '/archives',
    },
    {
      text: 'Tags',
      link: '/tags',
    },
    {
      text: 'Github',
      link: '/github',
    },
    {
      text: 'Weibo',
      link: '/weibo',
    },
  ];

  const LeftNav = () => <Nav navItems={navLinkItems} />;

  const store = new SkeletonStore();
  const Header = createHeader({
    onCloseClicked: () => store.hideMenu(),
    onMenuClicked: () => store.showMenu(),
  });
  const SkeletonImpl = () => (
    <Skeleton
      Header={Header}
      LeftNav={LeftNav}
      MainContent={MainContent}
      isMenuHidden={store.isMenuHidden}
    />
  );
  return observer(SkeletonImpl);
}
