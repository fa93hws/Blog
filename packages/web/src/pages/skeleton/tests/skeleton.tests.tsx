import * as React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { setupIntersectionObserverMock } from 'utils/tests/mock-intersection-observer';
import { Skeleton, createSkeleton } from '../skeleton';

describe('Skeleton', () => {
  const LeftNav = () => <div>Nav</div>;
  const MainContent = () => <div>MainContent</div>;
  const Header = () => <div>Header</div>;

  it('renders the skeleton when menu is hidden', () => {
    expect(
      <Skeleton
        LeftNav={LeftNav}
        MainContent={MainContent}
        Header={Header}
        isMenuHidden={false}
      />,
    ).toMatchRenderedSnapshot();
  });

  it('renders the skeleton when menu is open', () => {
    expect(
      <Skeleton
        LeftNav={LeftNav}
        MainContent={MainContent}
        Header={Header}
        isMenuHidden={true}
      />,
    ).toMatchRenderedSnapshot();
  });

  it('mount the skeleton without error', async () => {
    setupIntersectionObserverMock();
    const SkeletonImpl = createSkeleton();
    const mountPromise = act(async () => {
      mount(<SkeletonImpl />);
    });
    // linting is wrong
    // eslint-disable-next-line jest/valid-expect
    expect(mountPromise).resolves.toBe(undefined);
  });
});
