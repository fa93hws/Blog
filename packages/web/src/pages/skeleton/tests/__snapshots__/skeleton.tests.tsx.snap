// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton renders the skeleton when menu is hidden 1`] = `
Array [
  <aside
    class="nav"
  >
    <div>
      Nav
    </div>
  </aside>,
  <main
    class="mainContent"
  >
    <div>
      <div>
        Header
      </div>
      <div>
        MainContent
      </div>
    </div>
  </main>,
]
`;

exports[`Skeleton renders the skeleton when menu is open 1`] = `
Array [
  <aside
    class="nav hideMenu"
  >
    <div>
      Nav
    </div>
  </aside>,
  <main
    class="mainContent hideMenu"
  >
    <div>
      <div>
        Header
      </div>
      <div>
        MainContent
      </div>
    </div>
  </main>,
]
`;
